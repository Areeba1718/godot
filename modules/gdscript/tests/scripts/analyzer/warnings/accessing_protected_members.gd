class A:
	@warning_ignore("unused_private_class_variable")
	var _a
	static var _b

class B extends A:
	var a = A.new()
	var b = a._a
	@warning_ignore("accessing_private_member") var c = a._a
	var d = a._a
	var e = A._b
	@warning_ignore("accessing_private_member") var f = A._b
	var g = A._b

class C:
	var k = A.new()
	var l = k._a
	@warning_ignore("accessing_private_member") var m = k._a
	var n = k._a
	var o = A._b
	@warning_ignore("accessing_private_member") var p = A._b
	var q = A._b

func test():
	pass
