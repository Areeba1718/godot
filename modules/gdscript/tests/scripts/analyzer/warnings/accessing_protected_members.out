GDTEST_OK
>> WARNING
>> Line: 17
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "_a" in an external class.
>> WARNING
>> Line: 19
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "_a" in an external class.
>> WARNING
>> Line: 20
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "_b" in an external class.
>> WARNING
>> Line: 22
>> ACCESSING_PRIVATE_MEMBER
>> Trying accessing a private member "_b" in an external class.
